# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.h
# Commit: 37b28252e060
# Full Hash: 37b28252e06031f86628ad8197eb47fa3857b389
# Author: stransky <stransky@redhat.com>
# Date: 2022-09-19 15:58:06
# Regressor Bug: 1789581
# File Overlap Count: 1
# Description:
#   Bug 1789581 [Wayland] Use move-to-rect offset to shift popup position when there's a gap between popups r=emilio
#   
#   If there's a gap between popups we can't use move-to-rect as anchor needs to be placed inside of parent popup.
#   In such case place anchor to upper corner of parent popup and submit popup position as offset.
#   
# ==============================================================================

diff -r 540dcf23e80e -r 37b28252e060 widget/gtk/nsWindow.h
--- a/widget/gtk/nsWindow.h	Mon Sep 19 09:40:23 2022 +0000
+++ b/widget/gtk/nsWindow.h	Mon Sep 19 10:39:15 2022 +0000
@@ -846,8 +846,10 @@
   void WaylandPopupMove();
   void WaylandPopupPrepareForMove();
   bool WaylandPopupRemoveNegativePosition(int* aX = nullptr, int* aY = nullptr);
-  bool WaylandPopupCheckAndGetAnchor(GdkRectangle* aPopupAnchor);
-  bool WaylandPopupAnchorAdjustForParentPopup(GdkRectangle* aPopupAnchor);
+  bool WaylandPopupCheckAndGetAnchor(GdkRectangle* aPopupAnchor,
+                                     GdkPoint* aOffset);
+  bool WaylandPopupAnchorAdjustForParentPopup(GdkRectangle* aPopupAnchor,
+                                              GdkPoint* aOffset);
   nsWindow* WaylandPopupGetTopmostWindow();
   bool IsPopupInLayoutPopupChain(nsTArray<nsIWidget*>* aLayoutWidgetHierarchy,
                                  bool aMustMatchParent);
